### PR TITLE
Add VK_BASALT_EXPORT

### LINKED ISSUE #207: 0.3.2.8: fails to build with error: 'VK_LAYER_EXPORT' does not name a type
using
glslang: 12.0.0
vulkan: 1.3.241
```
FAILED: src/libvkbasalt.so.p/basalt.cpp.o 
x86_64-pc-linux-gnu-c++ -Isrc/libvkbasalt.so.p -Isrc -I../vkBasalt-0.3.2.8/src -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++2a -O0 -g -march=native -O2 -pipe -march=native -O2 -fPIC -MD -MQ src/libvkbasalt.so.p/basalt.cpp.o -MF src/libvkbasalt.so.p/basalt.cpp.o.d -o src/libvkbasalt.so.p/basalt.cpp.o -c ../vkBasalt-0.3.2.8/src/basalt.cpp
../vkBasalt-0.3.2.8/src/basalt.cpp:70:5: error: 'VK_LAYER_EXPORT' does not name a type
   70 |     VK_LAYER_EXPORT VkResult VKAPI_CALL vkBasalt_CreateInstance(const VkInstanceCreateInfo*  pCreateInfo,
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:136:5: error: 'VK_LAYER_EXPORT' does not name a type
  136 |     VK_LAYER_EXPORT void VKAPI_CALL vkBasalt_DestroyInstance(VkInstance instance, const VkAllocationCallbacks* pAllocator)
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:151:5: error: 'VK_LAYER_EXPORT' does not name a type
  151 |     VK_LAYER_EXPORT VkResult VKAPI_CALL vkBasalt_CreateDevice(VkPhysicalDevice             physicalDevice,
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:286:5: error: 'VK_LAYER_EXPORT' does not name a type
  286 |     VK_LAYER_EXPORT void VKAPI_CALL vkBasalt_DestroyDevice(VkDevice device, const VkAllocationCallbacks* pAllocator)
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:750:5: error: 'VK_LAYER_EXPORT' does not name a type
  750 |     VK_LAYER_EXPORT VkResult VKAPI_CALL vkBasalt_EnumerateInstanceLayerProperties(uint32_t* pPropertyCount, VkLayerProperties* pProperties)
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:766:5: error: 'VK_LAYER_EXPORT' does not name a type
  766 |     VK_LAYER_EXPORT VkResult VKAPI_CALL vkBasalt_EnumerateDeviceLayerProperties(VkPhysicalDevice   physicalDevice,
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:773:5: error: 'VK_LAYER_EXPORT' does not name a type
  773 |     VK_LAYER_EXPORT VkResult VKAPI_CALL vkBasalt_EnumerateInstanceExtensionProperties(const char*            pLayerName,
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:790:5: error: 'VK_LAYER_EXPORT' does not name a type
  790 |     VK_LAYER_EXPORT VkResult VKAPI_CALL vkBasalt_EnumerateDeviceExtensionProperties(VkPhysicalDevice       physicalDevice,
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:820:5: error: 'VK_LAYER_EXPORT' does not name a type
  820 |     VK_LAYER_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetDeviceProcAddr(VkDevice device, const char* pName);
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:821:5: error: 'VK_LAYER_EXPORT' does not name a type
  821 |     VK_LAYER_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetInstanceProcAddr(VkInstance instance, const char* pName);
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:860:5: error: 'VK_LAYER_EXPORT' does not name a type
  860 |     VK_LAYER_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetDeviceProcAddr(VkDevice device, const char* pName)
      |     ^~~~~~~~~~~~~~~
../vkBasalt-0.3.2.8/src/basalt.cpp:875:5: error: 'VK_LAYER_EXPORT' does not name a type
  875 |     VK_LAYER_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetInstanceProcAddr(VkInstance instance, const char* pName)
      |     ^~~~~~~~~~~~~~~
```
full build log: [vkbasalt-0.3.2.8.log.txt](https://github.com/DadSchoorse/vkBasalt/files/10828612/vkbasalt-0.3.2.8.log.txt)

### PR DESCRIPTION
**Pull request by juan-lunarg**
closes #207

### CODE REVIEW

File: src/basalt.cpp [hunk 1]
```diff
@@ -817,8 +823,8 @@ namespace vkBasalt
 extern "C"
 { // these are the entry points for the layer, so they need to be c-linkeable
 
-    VK_LAYER_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetDeviceProcAddr(VkDevice device, const char* pName);
-    VK_LAYER_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetInstanceProcAddr(VkInstance instance, const char* pName);
+    VK_BASALT_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetDeviceProcAddr(VkDevice device, const char* pName);
+    VK_BASALT_EXPORT PFN_vkVoidFunction VKAPI_CALL vkBasalt_GetInstanceProcAddr(VkInstance instance, const char* pName);
```
Comment (juan-lunarg) [inline src/basalt.cpp:827]: Note. I'm almost positive only these 2 functions need to be exported based on the `vkBasalt.json.in`

File: src/meson.build [hunk 2]
```diff
@@ -56,5 +56,6 @@ shared_library(meson.project_name().to_lower(),
     link_with: [keyboard_input_x11_lib],
     include_directories : vkBasalt_include_path,
     dependencies : [x11_dep, reshade_dep],
+    gnu_symbol_visibility: 'hidden'
```
Comment (juan-lunarg) [inline src/meson.build:59]: By default GNU exports all symbols (one of the rare cases MSVC actually has the better default):
https://mesonbuild.com/Release-notes-for-0-48-0.html#keyword-argument-for-gnu-symbol-visibility

Comment (tgurr) [reply src/meson.build:59]: The line is missing a `,` at the end creating a meson build error:

```
../vkBasalt-0.3.2.8/src/meson.build:60:4: ERROR: Expecting rparen got id.
    install : true,
    ^
For a block that started at 54,14
shared_library(meson.project_name().to_lower(),
```

Comment (juan-lunarg) [reply src/meson.build]: Fixed.

**Comment by juan-lunarg** [comment]
@DadSchoorse does the PR look good to you?

**Comment by DadSchoorse** [comment]
Thanks for the fix!

File: src/meson.build [hunk 3]
```diff
@@ -56,5 +56,6 @@ shared_library(meson.project_name().to_lower(),
     link_with: [keyboard_input_x11_lib],
     include_directories : vkBasalt_include_path,
     dependencies : [x11_dep, reshade_dep],
+    gnu_symbol_visibility: 'hidden',
```
Comment (juan-lunarg) [inline src/meson.build:59]: This little bit here will also help with the size of the binary fwiw

**Pull request merged**